### PR TITLE
Boot snapshots support

### PR DESCRIPTION
**Pull request by BShrivastav**
This feature adds support to provision images using boot snapshot id.

- Mutual exclusivity with boot volume, catalog and image is maintained with this attribute.

### CODE REVIEW

File: developer/examples/build.vpc.boot.snapshot.pkr.hcl [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+//   required_plugins {
+//     ibmcloud = {
+//       version = ">=v3.0.0"
+//       source = "github.com/IBM/ibmcloud"
+//     }
+//   }
+// }
+
+variable "IBM_API_KEY" {
+  type = string
```
Comment (SunithaGudisagarIBM1) [inline developer/examples/build.vpc.boot.snapshot.pkr.hcl:11]: Hi Bhavesh, 

I ran make run-boot-snapshot and i am getting the attached error though there is no "vsi_boot_volume_id" specified in the example, can you please run this example and attach the result in the PR.

Thank you

File: developer/examples/build.vpc.boot.snapshot.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+variable "SECURITY_GROUP_ID" {
+  type = string
+}
+
+variable "BOOT_SNAPSHOT_ID" {
+  type = string
+}
+
+// variable "VPC_URL" {
```
Comment (SunithaGudisagarIBM1) [inline developer/examples/build.vpc.boot.snapshot.pkr.hcl:34]: Remove this if not required

File: examples/build.vpc.boot.snapshot.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+variable "SECURITY_GROUP_ID" {
+  type = string
+}
+
+variable "BOOT_SNAPSHOT_ID" {
+  type = string
+}
+
+// variable "VPC_URL" {
```
Comment (SunithaGudisagarIBM1) [inline examples/build.vpc.boot.snapshot.pkr.hcl:34]: remove if not required

File: builder/ibmcloud/vpc/config.go [hunk 4]
```diff
@@ -85,21 +86,46 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, error) {
...
 	}
+	if c.VSIBootSnapshotID != "" {
+		oneOfInput = oneOfInput + 1
+	}
+
+	if oneOfInput != 1 {
+		errs = packer.MultiErrorAppend(errs, errors.New("only one of (vsi_base_image_id or vsi_base_image_name) or (catalog_offering_crn or catalog_offering_version_crn) or vsi_boot_volume_id or vsi_boot_snapshot_id is required"))
+	}
+
+	// if (c.CatalogOfferingCRN != "" || c.CatalogOfferingVersionCRN != "") && (c.VSIBaseImageID != "" || c.VSIBaseImageName != "") && c.VSIBootVolumeID != "" && c.VSIBootSnapshotID != "" {
```
Comment (SunithaGudisagarIBM1) [inline builder/ibmcloud/vpc/config.go:114]: Can you please remove this commented lines if not required..

Thank you

File: developer/examples/build.vpc.boot.snapshot.pkr.hcl (hunk 1 above)
Comment (SunithaGudisagarIBM1) [reply developer/examples/build.vpc.boot.snapshot.pkr.hcl:11]: Screenshot 2023-06-04 at 1 44 07 PM

**Review by SunithaGudisagarIBM1 — CHANGES_REQUESTED** [review]
Hi Bhavesh,

I ran make run-boot-snapshot and i am getting the attached error though there is no "vsi_boot_volume_id" specified in the example, can you please run this example and attach the result in the PR.

Thank you

Screenshot 2023-06-04 at 1 44 07 PM

File: builder/ibmcloud/vpc/config.go (hunk 4 above)
Comment (BShrivastav) [reply builder/ibmcloud/vpc/config.go:114]: Logic changes, seeking review from other dev stake holders on this, will clear ones done.

File: examples/build.vpc.boot.snapshot.pkr.hcl (hunk 3 above)
Comment (BShrivastav) [reply examples/build.vpc.boot.snapshot.pkr.hcl:34]: This is for stage.

File: developer/examples/build.vpc.boot.snapshot.pkr.hcl (hunk 2 above)
Comment (BShrivastav) [reply developer/examples/build.vpc.boot.snapshot.pkr.hcl:34]: optional stage support.

File: developer/examples/build.vpc.boot.snapshot.pkr.hcl (hunk 1 above)
Comment (BShrivastav) [reply developer/examples/build.vpc.boot.snapshot.pkr.hcl:11]: hey, it is cause the latest changes were built at your end, please follow readme:

```
import go env...
make setup
make plugin
```

**Comment by SunithaGudisagarIBM1** [comment]
Testing done, PR looks good

**Comment by astha-jain** [comment]
@ujjwal-ibm Could you please cleanup this PR?

File: examples/build.vpc.boot.snapshot.pkr.hcl (hunk 3 above)
Comment (ujjwal-ibm) [reply examples/build.vpc.boot.snapshot.pkr.hcl:34]: removed

File: builder/ibmcloud/vpc/config.go (hunk 4 above)
Comment (ujjwal-ibm) [reply builder/ibmcloud/vpc/config.go:123]: removed

File: developer/examples/build.vpc.boot.snapshot.pkr.hcl (hunk 2 above)
Comment (ujjwal-ibm) [reply developer/examples/build.vpc.boot.snapshot.pkr.hcl:34]: removed

**Review by astha-jain — APPROVED** [review]

**Pull request merged**